### PR TITLE
Refactoring and fixing small issues in service indicator

### PR DESCRIPTION
**Pull request by dkostic**
### Issues:
N/A

### Description of changes: 
Refactoring and fixing small issues in service indicator

### Call-outs:
Point out areas that need special attention or support during the review process. Discuss architecture or design changes.

### Testing:
How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (samuel40791765) [inline crypto/fipsmodule/service_indicator/service_indicator.c:69]: I remember that we did `(before != after)` to avoid `after` overflowing and becoming less than `before`

Comment (dkostic) [reply crypto/fipsmodule/service_indicator/service_indicator.c:69]: can we then remove the check in line 88, in update_state function? If the check is `before != after` why do we have a counter then, instead of a `{0, 1}` flag?

Comment (samuel40791765) [reply crypto/fipsmodule/service_indicator/service_indicator.c:69]: Yeah, now that you point that out, line 88 was a recent change when we were trying to fix underflow/overflow errors with the service indicator locks. After our internal discussion, I think it's alright to do `before < after` if we're checking for overflowing there.

**Review by samuel40791765 — DISMISSED** [review]

**Review by DominicDams — APPROVED** [review]

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [reply crypto/fipsmodule/service_indicator/service_indicator.c:69]: I'm now worried about a service that has used more than 2^64 FIPS operations since its start, wouldn't the check on l.87 fail when the counter = 2^{64} - 1? I think that's why we didn't have that check and we were only checking if the value changed. (Having that check on the lock makes sense because our call graph is few levels deep.)

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:169]: I'm afraid to make those changes now before we work on the static build.

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:188]: same comment as related to the delocator.

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:229]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:239]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:255]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:268]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:306]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:311]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:347]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:285]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:329]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:389]: Nit: remove the blank line?

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [inline crypto/fipsmodule/service_indicator/service_indicator.c:390]: I hope the delocator will not complain.

File: crypto/fipsmodule/ec/ec_key.c
Comment (nebeid) [inline crypto/fipsmodule/ec/ec_key.c:424]: Just curious: why was the cast needed here and not in the other change?

**Review by nebeid — COMMENTED** [review]
Mostly "nits" about blank lines and continuing the discussion on updating/checking the counter.

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (dkostic) [reply crypto/fipsmodule/service_indicator/service_indicator.c:69]: yes, our current design allows only 2^64 FIPS approved operations, after that, it would always indicate "not approved". That's why my suggestion is to remove the check in line 87 and keep the `before != after` condition. This way we don't have the artificial limit on the number of FIPS approved operations and we don't have to care about overflowing the counter.

p.s. the counters are thread local so assuming that one FIPS approved operation takes 1 ns, it will take almost 600 years to overflow the counter :)

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (dkostic) [reply crypto/fipsmodule/service_indicator/service_indicator.c:169]: don't worry, I'll start working on the static build right after this so I'll change it back if need be :) The thing is, I really don't see how this has anything to do with delocator, so I'm hoping it was some other issue causing delocation to fail.

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (dkostic) [reply crypto/fipsmodule/service_indicator/service_indicator.c:229]: I added all the blank lines because to me it's more readable that way. If for you it's better without the blank lines I'll remove them?

File: crypto/fipsmodule/ec/ec_key.c
Comment (dkostic) [reply crypto/fipsmodule/ec/ec_key.c:424]: the function argument in this function is declared as `const` so the compiler complains that we are dropping the const qualifier.

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [reply crypto/fipsmodule/service_indicator/service_indicator.c:69]: I agree with you, Dusan in keeping the condition as `!=` and removing the check in l.87.
Good observation and the calculation :)

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (nebeid) [reply crypto/fipsmodule/service_indicator/service_indicator.c:229]: It's just different from the style in all other files and I would personally prefer the no line after the function signature. Also here, some of the `if` statements have their first line followed by a blank lines and some don't.

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (dkostic) [reply crypto/fipsmodule/service_indicator/service_indicator.c]: done

File: crypto/fipsmodule/service_indicator/service_indicator.c
Comment (dkostic) [reply crypto/fipsmodule/service_indicator/service_indicator.c]: ok, I removed the blank lines after function signatures and after if statements.

**Review by nebeid — APPROVED** [review]

**Review by samuel40791765 — APPROVED** [review]

**Pull request merged**
